test/Microsoft.Framework.Internal.Test/PropertyHelperTest.cs

```diff
@@ -343,6 +343,102 @@ public void MakeFastPropertySetter_SetsPropertyValues_ForNewedProperties()
             Assert.Equal("NewedTest value", instance.PropB);
         }
 
+        [Fact]
+        public void MakeFastPropertyGetter_ReferenceType_ForNullObject_Throws()
+        {
+            // Arrange
+            var property = PropertyHelper
+                .GetProperties(typeof(BaseClass))
+                .Single(p => p.Name == nameof(BaseClass.PropA));
+
+            var accessor = PropertyHelper.MakeFastPropertyGetter(property.Property);
+
+            // Act & Assert
+            Assert.Throws<NullReferenceException>(() => accessor(null));
+        }
+
+        [Fact]
+        public void MakeFastPropertyGetter_ValueType_ForNullObject_Throws()
+        {
+            // Arrange
+            var property = PropertyHelper
+                .GetProperties(typeof(MyProperties))
+                .Single(p => p.Name == nameof(MyProperties.StringProp));
+
+            var accessor = PropertyHelper.MakeFastPropertyGetter(property.Property);
+
+            // Act & Assert
+            Assert.Throws<NullReferenceException>(() => accessor(null));
+        }
+
+        [Fact]
+        public void MakeNullSafeFastPropertyGetter_ReferenceType()
+        {
+            // Arrange
+            var property = PropertyHelper
+                .GetProperties(typeof(BaseClass))
+                .Single(p => p.Name == nameof(BaseClass.PropA));
+
+            var accessor = PropertyHelper.MakeNullSafeFastPropertyGetter(property.Property);
+
+            // Act
+            var value = accessor(new BaseClass() { PropA = "Hi" });
+
+            // Assert
+            Assert.Equal("Hi", value);
+        }
+
+        [Fact]
+        public void MakeNullSafeFastPropertyGetter_ValueType()
```

Comment:
nit: for this and the above method, you've described the scenario but not the expectation.  perhaps `..._Succeeds` is enough